magenta-lib/src/main/scala/magenta/tasks/FastlyComputeTasks.scala

```diff
@@ -23,38 +24,69 @@ case class FastlyComputeTasks(s3Package: S3Path)(implicit
   implicit val ec: ExecutionContextExecutorService =
     ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(10))
 
+  private def _execute(
+      client: FastlyApiClient,
+      resources: DeploymentResources,
+      stopFlag: => Boolean
+  ): Try[Int] = {
+    for {
+      activeVersionNumber <-
+        Try(getActiveVersionNumber(client, resources.reporter, stopFlag))
+      nextVersionNumber <-
+        Try(clone(activeVersionNumber, client, resources.reporter, stopFlag))
+      _ <-
+        Try(
+          uploadPackage(
+            nextVersionNumber,
+            s3Package,
+            client,
+            resources.reporter,
+            stopFlag
+          )
+        )
+      _ <-
+        Try(
+          commentVersion(
+            nextVersionNumber,
+            client,
+            resources.reporter,
+            parameters,
+            stopFlag
+          )
+        )
+      _ <-
+        Try(
+          activateVersion(
+            nextVersionNumber,
+            client,
+            resources.reporter,
+            stopFlag
+          )
+        )
+    } yield nextVersionNumber
+  }
+
   override def execute(
       resources: DeploymentResources,
       stopFlag: => Boolean
   ): Unit = {
-    FastlyApiClientProvider.get(keyRing).foreach { client =>
-      val activeVersionNumber =
-        getActiveVersionNumber(client, resources.reporter, stopFlag)
-      val nextVersionNumber =
-        clone(activeVersionNumber, client, resources.reporter, stopFlag)
-
-      uploadPackage(
-        nextVersionNumber,
-        s3Package,
-        client,
-        resources.reporter,
-        stopFlag
-      )
-
-      commentVersion(
-        nextVersionNumber,
-        client,
-        resources.reporter,
-        parameters,
-        stopFlag
-      )
-
-      activateVersion(nextVersionNumber, client, resources.reporter, stopFlag)
-
-      resources.reporter
-        .info(
-          s"Fastly Compute@Edge service ${client.serviceId} - version $nextVersionNumber is now active"
+    FastlyApiClientProvider.get(keyRing) match {
+      case None =>
```

Comment:
Similar flattening of pattern match could be done as for FastlyTask.scala

Reply:
Thanks - yes, I'm not a big fan of nested pattern matching either. Have refactored accordingly :)